src/getters.jl

```diff
@@ -1481,4 +1481,87 @@ function GetDrugAmounts(
 
 end
 
-export GetDatabasePersonIDs, GetPatientState, GetPatientGender, GetPatientRace, GetPatientAgeGroup, GetPatientVisits, GetMostRecentConditions, GetMostRecentVisit, GetVisitCondition, GetPatientEthnicity, GetDatabaseYearRange, GetVisitPlaceOfService, GetVisitConcept, GetVisitDate, GetDrugExposures, GetDrugConceptIDs, GetDrugAmounts
+"""
+GetVisitProcedure(visit_ids, conn; tab = procedure_occurrence)
+
+Given a list of visit IDs, find their corresponding procedures.
+
+# Arguments:
+
+- `visit_ids` - list of `visit_id`'s; each ID must be of subtype `Integer`
+
+- `conn` - database connection using DBInterface
+
+# Keyword Arguments:
+
+- `tab` - the `SQLTable` representing the Condition Occurrence table; default `procedure_occurrence`
+
+# Returns
+
+- `df::DataFrame` - a two column `DataFrame` comprised of columns: `:visit_occurrence_id` and `:procedure_concept_id`
+"""
+function GetVisitProcedure(
+    visit_ids,
+    conn;
+    tab=procedure_occurrence
+)
+
+    df = DBInterface.execute(conn, GetVisitProcedure(visit_ids; tab=tab)) |> DataFrame
+
+    return df
+
+end
+
+"""
+function GetVisitProcedure(df:DataFrame, conn; tab = procedure_occurrence)
+
+Given a `DataFrame` with a `:visit_occurrence_id` column, return the `DataFrame` with an associated `:procedure_concept_id` for each `visit_occurrence_id` in the `DataFrame`
+
+Multiple dispatch that accepts all other arguments like in `GetVisitProcedure(ids, conn; tab = procedure_occurrence)`
+"""
+
+function GetVisitProcedure(
+    df::DataFrame,
+    conn;
+    tab=procedure_occurrence
+)
+
+    df_ids= df[:,"procedure_occurrence_id"]
+    
+    return outerjoin(GetVisitProcedure(df_ids, conn; tab=tab), df, on = :procedure_occurrence_id)
+
+end
```

Comment:
Almost correct! So, on line 1529, we actually are expecting the DataFrame to index for "visit_occurrence_id" (you'd also change this in line 1531). Additionally, you need to test passing in a dataframe to this dispatch of the function (if you see the test suite, I did a little bit of tweaking to add in a `test_df` variable that you could use). Thanks! :smile: